bots/store.js

```diff
@@ -0,0 +1,52 @@
+const { createSelector } = require('redux-bundler')
+const { merge } = require('ramda')
+
+module.exports = {
+  name: 'bots',
+  getReducer: function () {
+    const initialState = {
+      data: null,
+      isLoading: false,
+      error: null
+    }
+
+    return (state = initialState, { type, bot, error }) => {
+      if (type === 'FIND_BOTS_STARTED') {
+        return merge(state, { isLoading: true })
+      } else if (type === 'FIND_BOTS_FINISHED') {
+        return merge(state, {
+          data: merge(state.data, { [bot.botId]: bot }),
+          error: null,
+          isLoading: false
+        })
+      } else if (type === 'FIND_BOTS_FAILED') {
+        return merge(state, {
+          error,
+          isLoading: false
+        })
+      }
+      return state
+    }
+  },
+  doFindBots: data => ({ dispatch, client }) => {
+    dispatch({ type: 'FIND_BOTS_STARTED' })
+    client
+      .service('bot')
+      .find(data)
+      .then(bots => {
+        dispatch({ type: 'FIND_BOTS_FINISHED', bots })
+      })
+      .catch(error => {
+        dispatch({ type: 'FIND_BOTS_FAILED', error })
+      })
+  },
+  doCreateBot: botId => {},
+  selectBotsData: state => state.bots.data,
```

Comment:
will want selector for `selectPubsByUserId` (using `lodash` `groupBy` `userId`) and then using this for a `selectAuthenticatedUserPubs` (using `selectAuthenticatedUser`)

Reply:
might have read this too quickly the first time. wouldn't this be accomplished on the server side already since the `find` method should only return the pubs belonging to a certain user?